packages/zwave-js/src/lib/commandclass/HumidityControlSetpointCC.ts

```diff
@@ -0,0 +1,937 @@
+	// We expect a Thermostat Setpoint Report that matches the requested setpoint type
+	return received.type === sent.setpointType;
+}
+
+interface HumidityControlSetpointCCGetOptions extends CCCommandOptions {
+	setpointType: HumidityControlSetpointType;
+}
+
+@CCCommand(HumidityControlSetpointCommand.Get)
+@expectedCCResponse(
+	HumidityControlSetpointCCReport,
+	testResponseForHumidityControlSetpointGet,
+)
+export class HumidityControlSetpointCCGet extends HumidityControlSetpointCC {
+	public constructor(
+		driver: Driver,
+		options:
+			| CommandClassDeserializationOptions
+			| HumidityControlSetpointCCGetOptions,
+	) {
+		super(driver, options);
+		if (gotDeserializationOptions(options)) {
+			// TODO: Deserialize payload
+			throw new ZWaveError(
+				`${this.constructor.name}: deserialization not implemented`,
+				ZWaveErrorCodes.Deserialization_NotImplemented,
+			);
+		} else {
+			this.setpointType = options.setpointType;
+		}
+	}
+
+	public setpointType: HumidityControlSetpointType;
+
+	public serialize(): Buffer {
+		this.payload = Buffer.from([this.setpointType & 0b1111]);
+		return super.serialize();
+	}
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		return {
+			...super.toLogEntry(),
+			message: {
+				"setpoint type": getEnumMemberName(
+					HumidityControlSetpointType,
+					this.setpointType,
+				),
+			},
+		};
+	}
+}
+
+@CCCommand(HumidityControlSetpointCommand.SupportedReport)
+export class HumidityControlSetpointCCSupportedReport extends HumidityControlSetpointCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		validatePayload(this.payload.length >= 1);
+		this._supportedSetpointTypes = parseBitMask(
+			this.payload,
+			HumidityControlSetpointType["N/A"],
+		);
+
+		this.persistValues();
+	}
+
+	private _supportedSetpointTypes: HumidityControlSetpointType[];
+	@ccValue({ internal: true })
+	public get supportedSetpointTypes(): readonly HumidityControlSetpointType[] {
+		return this._supportedSetpointTypes;
+	}
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		return {
+			...super.toLogEntry(),
+			message: {
+				"supported setpoint types": this.supportedSetpointTypes
+					.map(
+						(t) =>
+							`\n· ${getEnumMemberName(
+								HumidityControlSetpointType,
+								t,
+							)}`,
+					)
+					.join(""),
+			},
+		};
+	}
+}
+
+@CCCommand(HumidityControlSetpointCommand.SupportedGet)
+@expectedCCResponse(HumidityControlSetpointCCSupportedReport)
+export class HumidityControlSetpointCCSupportedGet extends HumidityControlSetpointCC {}
+
+@CCCommand(HumidityControlSetpointCommand.ScaleSupportedReport)
+export class HumidityControlSetpointCCScaleSupportedReport extends HumidityControlSetpointCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		validatePayload(this.payload.length >= 1);
+
+		const supportedScaleIndices = parseBitMask(this.payload, 0);
```

Comment:
just a note on this command: most of the commands that pertain to a single setpoint will include the setpoint in the response and I noticed there are checks to ensure the response is for the same setpoint (e.g. `testResponseForThermostatSetpointGet`. This response doesn't include the `setpointType` per the spec, and that comes with a few caveats:
1) not possible to ensure the response is for the correct setpoint
2) not possible to save the supported scales for the given setpoint as `states` metadata on the `setpointScale` value, so I do it in `interview()`
3) log message won't mention the setpoint type like the other ones do

Reply:
Yay, consistent specs 🎉